cmd/oceantv/broadcast_hardware_machine.go

```diff
@@ -32,6 +32,16 @@ func newHardwareStarting(ctx *broadcastContext) *hardwareStarting {
 }
 func (s *hardwareStarting) enter() {
 	s.LastEntered = time.Now()
+	if s.cfg.ControllerMAC != 0 {
+		controllerIsOn, err := getDeviceStatus(context.Background(), s.cfg.ControllerMAC, settingsStore)
+		if err != nil {
+			disableBroadcast(s.cfg, 1, fmt.Errorf("failed to get controller status: %v", err))
```

Comment:
So I'm thinking in both of the error cases we should instead publish a controllerFailureEvent (doesn't exist yet). We'll add a handler for this in this state machine, which we then use to transition from this state to hardwareOff. The broadcast state machine will also handle this event type and there we transition to the idle state for secondary and direct broadcasts (permanent broadcasts should go to slate). That's also where we can call onFailureClosure (my bad for suggesting we factor out disableBroadcast; that could work, but now I'm thinking onFailureClosure could take another parameter that can be used to ignore maxFailures?) Thoughts apprciated.

Reply:
These seems reasonable, maybe instead of ignoring maxFailures, you can pass the number of failures you want to allow? This would make this value more configurable for different attempts?

ie controller failures only get 1 attempt, youtube failures get 3, and hardware failures get 5. This would also make it easier to allow the user to specify this via cloudblue

Reply:
That could work, but what feels weird is that we don't have different issue tallies for each type of issue. Instead maybe we need an issue severity level or something, i.e. a factor that is multiplied by the issue tally increment i.e. we still have a consistent max limit but different issues will have different severities and go over it faster. Alternatively, we could just have two severity levels. One that increments the tally, and one that just fails completely immediately.

Reply:
This feels like maybe beyond the scope of this PR. Maybe for now, it should include a instant fail bool which skips the tally, and any additional changes can be implemented later?

Reply:
I have currently implemented a disable on first fail boolean argument which will allow failures to skip the count and fail immediately, and have written an issue to make this better implemented #144.